common/utils.go

```diff
@@ -491,6 +491,7 @@ func (meta *testMeta) startTestFactories(numFactories int, providerFn TestProvid
 		for i := 0; atomic.LoadInt64(&meta.abort) == 0; i++ {
 			if fileName, err := providerFn(i, threadId); err != nil {
 				log.Error("Error storing test", "err", err)
+				atomic.StoreInt64(&meta.abort, 1)
```

Comment:
I don't understand why this is needed. `log.Info("Last test factory exiting\n")` -- the last one should close the testchannel, and that should gracefully close the pipeline. 

I'm not saying it is not needed, just that we should figure out why the intended mechanism isn't working.

Reply:
Because when I ran it, I saw "Last test factory exiting" message, but it still didn't shut down, just kept incrementing the time and making the stats look slower and slower (though no more tests were being executed)